bundle/Command/ScheduledVisibilityUpdateCommand.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Command;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\Values\Content\Language;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Core\Configuration;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Core\Exception\InvalidStateException;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Core\ScheduledVisibilityService;
+use Pagerfanta\Doctrine\DBAL\QueryAdapter;
+use Pagerfanta\Pagerfanta;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+use function count;
+use function sprintf;
+
+final class ScheduledVisibilityUpdateCommand extends Command
+{
+    private SymfonyStyle $style;
+    private array $languageCache = [];
+
+    public function __construct(
+        private readonly Repository $repository,
+        private readonly ScheduledVisibilityService $scheduledVisibilityService,
+        private readonly Configuration $configurationService,
+        private readonly Connection $connection,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription(
+            'Updates content visibility based on publish_from and publish_to attributes and configuration.',
+        );
+        $this->addOption(
+            'limit',
+            'l',
+            InputOption::VALUE_OPTIONAL,
+            'Number of content object to process in a single iteration',
+            50,
```

Comment:
Since we now removed search from the picture, this can be increased quite a bit. I'll still stress test it, but we can probably increase it to 1024 without problems.

Reply:
fixed in 59a2b11254f6e58df7fff858c17b8a763c966b4d